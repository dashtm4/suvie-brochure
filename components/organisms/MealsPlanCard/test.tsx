import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { MealsPlanCard } from '.'

describe('<MealsPlanCard />', () => {
  const smartMealsPlan = {
    id: '1',
    title: 'Test title',
    description: 'Test description',
    price: {
      price: 10
    },
    note: 'Test note',
    includes: [{ item: 'Item 1' }, { item: 'Item 2' }],
    image: {
      url: 'image.jpg'
    },
    gallery: [
      {
        url: 'image.jpg'
      }
    ],
    recommended: false
  }

  it('should renders correct <MealsPlanCard />', () => {
    const wrapper = shallow(<MealsPlanCard smartMealsPlan={smartMealsPlan} />)

    expect(wrapper.contains(smartMealsPlan.title)).toBeTruthy()
    expect(wrapper.contains(smartMealsPlan.description)).toBeTruthy()
    expect(wrapper.contains(smartMealsPlan.price.price)).toBeTruthy()
    expect(wrapper.contains(smartMealsPlan.note)).toBeTruthy()
    expect(wrapper.contains('Most popular')).toBeFalsy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should display most popular badge', () => {
    const recommended = true

    const wrapper = shallow(
      <MealsPlanCard smartMealsPlan={{ ...smartMealsPlan, recommended }} />
    )

    expect(wrapper.contains('Most Popular')).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
