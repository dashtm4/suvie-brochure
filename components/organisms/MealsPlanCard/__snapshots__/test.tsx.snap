// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealsPlanCard /> should display most popular badge 1`] = `
<Fragment>
  <div
    className="flex flex-col border rounded-2xl cursor-pointer hover:border-primary-default overflow-hidden"
    onClick={[Function]}
  >
    <figure
      className="relative w-full h-40"
      role="none"
    >
      <Image
        alt="Test title"
        layout="fill"
        objectFit="cover"
        src="image.jpg"
      />
      <div
        className="absolute top-full left-1/2 py-1 px-2 font-serif text-lg font-normal text-white rounded-lg transform -translate-x-1/2 -translate-y-1/2 bg-secondary"
      >
        Most Popular
      </div>
    </figure>
    <div
      className="flex flex-col flex-grow pt-6 pb-5 mx-3 text-center"
    >
      <h3
        className="mb-2 font-serif text-3xl font-normal text-black"
      >
        Test title
      </h3>
      <div
        className="mb-4 text-quinary"
      >
        Test description
      </div>
      <div
        className="mt-auto mb-0"
      >
        <div
          className="mb-2 font-serif text-2xl font-normal text-secondary"
        >
          $
          10
          /serving
        </div>
        <div
          className="font-serif text-sm italic font-normal text-quinary"
        >
          Test note
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`<MealsPlanCard /> should renders correct <MealsPlanCard /> 1`] = `
<Fragment>
  <div
    className="flex flex-col border rounded-2xl cursor-pointer hover:border-primary-default overflow-hidden"
    onClick={[Function]}
  >
    <figure
      className="relative w-full h-40"
      role="none"
    >
      <Image
        alt="Test title"
        layout="fill"
        objectFit="cover"
        src="image.jpg"
      />
    </figure>
    <div
      className="flex flex-col flex-grow pt-6 pb-5 mx-3 text-center"
    >
      <h3
        className="mb-2 font-serif text-3xl font-normal text-black"
      >
        Test title
      </h3>
      <div
        className="mb-4 text-quinary"
      >
        Test description
      </div>
      <div
        className="mt-auto mb-0"
      >
        <div
          className="mb-2 font-serif text-2xl font-normal text-secondary"
        >
          $
          10
          /serving
        </div>
        <div
          className="font-serif text-sm italic font-normal text-quinary"
        >
          Test note
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
